src/tasks/selfSell.ts

```diff
@@ -787,11 +792,30 @@ export async function selfSell(input: SelfSellInput): Promise<string[] | null> {
     dryRun: input.dryRun,
     doNotPrompt: input.doNotPrompt,
   });
-  await submitSettlement({
-    ...input,
-    settlementContract: input.settlement,
-    encodedSettlement: finalSettlement,
-  });
+
+  if (!isSigner(input.solver)) {
```

Comment:
I'm not a huge fan of conflating `isSigner == !isSafe`. In fact, the `submitSettlement` already has some logic around whether or not the solver is a signer:

https://github.com/cowprotocol/contracts/blob/35625ec2392f02a24e9be2eea209f47dcbeb0db6/src/tasks/withdraw/settle.ts#L30-L38

I do think a flag for using a `--safe` might be more appropriate.